work/lessonbuilder_rewrite_urls.py

```diff
@@ -33,13 +34,21 @@ def run(SITE_ID, APP):
 
     sakai_url = APP['sakai_url']
     url_prefix = f"{sakai_url}/access/content/group/{SITE_ID}"
-
+    
     for item in root.findall(".//item[@type='5']"):
-        html_src = item.attrib['html']
-        if url_prefix in html_src:
-            html_src = html_src.replace(url_prefix, "..")
-            item.set('html', html_src)
-            rewrite = True
+
+        html = BeautifulSoup(item.attrib['html'], 'html.parser')
+        attr_list = ['src', 'href']
+        for attr in attr_list:
+            elements = html.find_all(attrs={attr: True})
+            for element in elements:
+                currenturl = element.get(attr)
+                if url_prefix in currenturl:
+                    updatedurl = currenturl.replace(sakai_url, "..").replace("%3A", '')
+                    element[attr] = updatedurl
+        
+                    item.set('html', str(html))
```

Comment:
Move this to after the for loop which starts in line 40 has completed. It should work fine as-is but it's calling item.set() multiple times possibly whereas it only needs to be updated once after all the elements have been updated.